src/QirTools/pyqir/pyqir/__init__.py

```diff
@@ -1,59 +1,214 @@
+        """
+        Applies measurement operation or the source qubit into the target register
+
+        :param qubit: name of the source qubit
+        :type qubit: str
+        :param target: name of the target register
+        :type target: str
+        """
         self.pyqir.m(qubit, target)
 
     def reset(self, target: str):
+        """
+        Applies Reset operation to the target qubit
+
+        :param target: name of the target qubit
+        :type target: str
+        """
         self.pyqir.reset(target)
 
     def rx(self, theta: float, qubit: str):
+        """
+        Applies Rx operation to the target qubit
+
+        :param theta: rotation value for target qubit
+        :type theta: float
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.rx(theta, qubit)
 
     def ry(self, theta: float, qubit: str):
+        """
+        Applies Ry operation to the target qubit
+
+        :param theta: rotation value for target qubit
+        :type theta: float
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.ry(theta, qubit)
 
     def rz(self, theta: float, qubit: str):
+        """
+        Applies Rz operation to the target qubit
+
+        :param theta: rotation value for target qubit
+        :type theta: float
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.rz(theta, qubit)
 
     def s(self, qubit: str):
+        """
+        Applies S operation to the target qubit
+
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.s(qubit)
 
     def s_adj(self, qubit: str):
+        """
+        Applies SAdj operation to the target qubit
+
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.s_adj(qubit)
 
     def t(self, qubit: str):
+        """
+        Applies T operation to the target qubit
+
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.t(qubit)
 
     def t_adj(self, qubit: str):
+        """
+        Applies TAdj operation to the target qubit
+
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.t_adj(qubit)
 
     def x(self, qubit: str):
+        """
+        Applies X operation to the target qubit
+
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.x(qubit)
 
     def y(self, qubit: str):
+        """
+        Applies Y operation to the target qubit
+
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.y(qubit)
 
     def z(self, qubit: str):
+        """
+        Applies Z operation to the target qubit
+
+        :param qubit: name of the target qubit
+        :type qubit: str
+        """
         self.pyqir.z(qubit)
 
```

Comment:
I see a lot of intrinsics missing, along with their adjoints and controlled. Are we planning to add them later?

Reply:
I believe this is meant to map to the set of gates defined in the Base Profile draft here: https://github.com/microsoft/qsharp-language/blob/ageller/profile/Specifications/QIR/Base-Profile.md. Note that the set of gates is the same except for the Toffoli gate.

Reply:
I am targeting the base profile as @swernli mentioned. We may add some others, but to start everything off we stayed with just the base. The QIR output also uses the higher level methods instead of the `__quantum__qis__??__body` calls.

interop.rs lines 89-107
``` rust
// https://github.com/microsoft/qsharp-language/blob/ageller/profile/Specifications/QIR/Base-Profile.md
#[derive(Clone, Debug, PartialEq)]
pub enum Instruction {
    Cx(Controlled),
    Cz(Controlled),
    H(Single),
    M(Measured),
    Reset(Single),
    Rx(Rotated),
    Ry(Rotated),
    Rz(Rotated),
    S(Single),
    SAdj(Single),
    T(Single),
    TAdj(Single),
    X(Single),
    Y(Single),
    Z(Single),
}
```